File: 1869. Longer Contiguous Segments of Ones than Zeros/conts_seg.py
"""
     * Solved By: Gabriel E. Rodriguez Garcia
     * Date Completed: July 25, 2023
     * Exersise #1869
     * Exercise Name: Longer Contigoud Segments of Ones Than Zeros 
     * Difficulty: Easy
     * Time Complexity: O(n)
     * Space Complexity: O(1)
     * Language Used: Python
     * Tags: tring
"""
def checkZeroOnes(self, s: str) -> bool:
        best_one, best_zero, current_one, current_zero = 0, 0, 0, 0
        
        for i in s:
            if i == "1":
                current_zero = 0
                current_one += 1
            else:
                current_zero += 1
                current_one = 0
                
            best_one = max(best_one, current_one)
            best_zero = max(best_zero, current_zero)
        
        return best_one > best_zero

File: 1869. Longer Contiguous Segments of Ones than Zeros/conts_seg.java
public class conts_seg {
/**
     * Solved By: Gabriel E. Rodriguez Garcia
     * Date Completed: April 14, 2022
     * Exersise #1869
     * Exercise Name: Longer Contigoud Segments of Ones Than Zeros 
     * Difficulty: Easy
     * Time Complexity: O(n)
     * Space Complexity: O(1)
     * Language Used: Java
     * Tags: String
*/
    public boolean checkZeroOnes(String s) {
        int max1 = 0;
        int max0 = 0;
        int count1 =0;
        int count0 =0;

        for(int i =0; i < s.length(); i++){
            char c = s.charAt(i);
            if(c=='0'){
               count0++;
               max1 = Math.max(count1, max1);
               count1=0;

            }else{
                count1++;
                max0 = Math.max(count0,max0);
                count0 =0;
            }

        }

        max1 = Math.max(max1, count1);
        max0 = Math.max(max0, count0);

        return max1 > max0;
    }
}